src/components/test/ConnectionsListProfiles.test.tsx

```diff
@@ -9,55 +9,147 @@ enum ListStatus {
 }
 
-const mockTempUserList = [
+const mockUserList = [
   preFabProfiles[0],
   preFabProfiles[1],
   preFabProfiles[2],
   preFabProfiles[3],
 ];
 
+const profiles = mockUserList.reduce((m, p) => ({ ...m, [p.fromId]: p }), {});
+
+const following = mockUserList
+  .slice(0, 2)
+  .reduce((m, p) => ({ ...m, [p.fromId]: true }), {});
+
+const followers = mockUserList
+  .slice(1)
+  .reduce((m, p) => ({ ...m, [p.fromId]: true }), {});
+
+const store = {
+  profiles: { profiles: profiles },
+};
+
 describe("ConnectionsListProfiles", () => {
   it("renders without crashing", () => {
     expect(() => {
       shallow(
-        componentWithStore(ConnectionsListProfiles, createMockStore({}), {
+        componentWithStore(ConnectionsListProfiles, createMockStore(store), {
           listStatus: ListStatus.FOLLOWING,
-          connectionsList: mockTempUserList,
-          notFollowingList: [preFabProfiles[1]],
+          followers: followers,
+          following: following,
         })
       );
     }).not.toThrow();
   });
 
-  it("displays correct follow button", () => {
-    const component = mount(
-      componentWithStore(ConnectionsListProfiles, createMockStore({}), {
-        listStatus: ListStatus.FOLLOWERS,
-        connectionsList: mockTempUserList,
-        notFollowingList: [preFabProfiles[0]],
-      })
-    );
-    expect(
-      component.find(".ConnectionsListProfiles__button").first().text()
-    ).toContain("Follow");
+  describe("when in closed mode", () => {
+    it("hides all connections", () => {
+      const component = mount(
+        componentWithStore(ConnectionsListProfiles, createMockStore(store), {
+          listStatus: ListStatus.CLOSED,
+          followers: followers,
+          following: following,
+        })
+      );
+      expect(component.find(".ConnectionsListProfiles__profile").length).toBe(
+        0
+      );
+    });
   });
 
-  it("displays correct unfollow button", () => {
-    const component = mount(
-      componentWithStore(ConnectionsListProfiles, createMockStore({}), {
-        listStatus: ListStatus.FOLLOWERS,
-        connectionsList: mockTempUserList,
-        notFollowingList: [preFabProfiles[0]],
-      })
-    );
-    expect(
-      component.find(".ConnectionsListProfiles__button").at(2).text()
-    ).toContain("Unfollow");
-    expect(
-      component.find(".ConnectionsListProfiles__button").at(3).text()
-    ).toContain("Unfollow");
-    expect(
-      component.find(".ConnectionsListProfiles__button").at(4).text()
-    ).toContain("Unfollow");
+  describe("when in following mode", () => {
+    it("shows all connections user is following", () => {
+      const component = mount(
+        componentWithStore(ConnectionsListProfiles, createMockStore(store), {
+          listStatus: ListStatus.FOLLOWING,
+          followers: followers,
+          following: following,
+        })
+      );
+      expect(component.find(".ConnectionsListProfiles__profile").length).toBe(
+        2
```

Comment:
I'd feel better if this compared with following length.  If there's a future bug (e.g. an off-by-1) and somehow this  list got changed this test could possibly pass.  Same with followers.

Reply:
Ok. next PR.